src/main/java/org/terasology/staticCities/sites/SiteFacetProvider.java

```diff
@@ -96,8 +107,8 @@ public void process(GeneratingRegion region) {
         // this can happen if other settlements further away with higher priority intersect
         for (Site site : sites) {
             float borderDist = config.maxRadius + config.minDistance + site.getRadius();
-            Rect2i certainRect = worldRect.expand(new Vector2i(-borderDist, -borderDist));
-            if (certainRect.contains(site.getPos())) {
+            BlockAreac certainRect = worldRect.expand(new Vector2i(new Vector2f(-borderDist, -borderDist), RoundingMode.FLOOR), new BlockArea(BlockArea.INVALID));
+            if (certainRect.contains(JomlUtil.from(site.getPos()))) {
                 settlementFacet.addSettlement(site);
             }
```

Comment:
I'd update this part by 

- reusing a single `tmpArea` to not create new areas for each site, as we only use it for checking the location
- already round the `borderDist` on computation, instead of creating to vectors for that

```suggestion
		BlockArea tmpArea = new BlockArea(BlockArea.INVALID);
        for (Site site : sites) {
            int borderDist = Math.roundUsing(config.maxRadius + config.minDistance + site.getRadius(), RoundMode.FLOOR);
            tmpArea.set(worldRect).expand(-borderDist, -borderDist);
            if (tmpArea.contains(JomlUtil.from(site.getPos()))) {
                settlementFacet.addSettlement(site);
            }
```